src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+using Microsoft.Win32.SafeHandles;
+
+using Internal.Cryptography;
+
+using ErrorCode = Interop.NCrypt.ErrorCode;
+using AsymmetricPaddingMode = Interop.NCrypt.AsymmetricPaddingMode;
+using BCRYPT_OAEP_PADDING_INFO = Interop.BCrypt.BCRYPT_OAEP_PADDING_INFO;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class RSACng : RSA
+    {
+        /// <summary>
+        ///     Encrypts data using the public key.
+        /// </summary>
+        public override byte[] Encrypt(byte[] data, RSAEncryptionPadding padding)
+        {
+            unsafe
```

Comment:
The method it’s calling is unsafe. It’s just well-hidden because it’s inside a delegate parameter which has a void*.


From: Stephen Toub [mailto:notifications@github.com]
Sent: Thursday, August 6, 2015 12:12 PM
To: dotnet/corefx
Cc: Atsushi Kanamori
Subject: Re: [corefx] Implement System.Security.Cryptography.RSACng (#2654)


In src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs<https://github.com/dotnet/corefx/pull/2654#discussion_r36453390>:

> +using Internal.Cryptography;

> +

> +using ErrorCode = Interop.NCrypt.ErrorCode;

> +using AsymmetricPaddingMode = Interop.NCrypt.AsymmetricPaddingMode;

> +using BCRYPT_OAEP_PADDING_INFO = Interop.BCrypt.BCRYPT_OAEP_PADDING_INFO;

> +

> +namespace System.Security.Cryptography

> +{

> +    public sealed partial class RSACng : RSA

> +    {

> +        /// <summary>

> +        ///     Encrypts data using the public key.

> +        /// </summary>

> +        public override byte[] Encrypt(byte[] data, RSAEncryptionPadding padding)

> +        {

> +            unsafe

Why is unsafe needed here and in the next function?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/2654/files#r36453390>.